datascience/tables.py

```diff
@@ -444,9 +444,26 @@ def column_index(self, column_label):
         return self.column_labels.index(column_label)
 
     def apply(self, fn, column_label):
-        """Returns an array where fn is applied to each element
-        of a specified column."""
-        return np.array([fn(v) for v in self[column_label]])
+        """Returns an array where ``fn`` is applied to each set of elements
+        by row from the specified columns in ``column_label``.
+
+        >>> letter = ['a', 'b', 'c', 'z']
+        >>> count = [9, 3, 3, 1]
+        >>> points = [1, 2, 2, 10]
+        >>> t = Table([letter, count, points], ['letter', 'count', 'points'])
+        >>> t
+        letter | count | points
+        a      | 9     | 1
+        b      | 3     | 2
+        c      | 3     | 2
+        z      | 1     | 10
+        >>> t.apply(lambda x, y: x * y, ['count', 'points'])
+        array([ 9,  6,  6, 10])
```

Comment:
Can we use the same documentation format as `Table.split`? Might as well leave the documentation better than how we found it when we can :)

It's important to mention that the `column_label` argument can take either a string or a list of column names.

In addition, we can throw an intelligent error when a column name passed in is not in the table — right now the student will only see a cryptic KeyError. See `Table.split` for an example.